crates/bevy_window/src/window.rs

```diff
@@ -138,6 +138,20 @@ pub struct Window {
     pub resolution: WindowResolution,
     /// Stores the title of the window.
     pub title: String,
+    /// Stores the application ID (on **`Wayland`**), `WM_CLASS` (on **`X11`**) or window class name (on **`Windows`**) of the window.
+    ///
+    /// For details about application ID conventions, see the [Desktop Entry Spec](https://specifications.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html#desktop-file-id).
+    /// For details about **`Windows`**'s window class names, see [About Window Classes](https://learn.microsoft.com/en-us/windows/win32/winmsg/about-window-classes).
+    ///
+    /// ## Platform-specific
+    ///
+    /// - **`Windows`**: Can only be set while building the window, setting the window's window class name.
+    /// - **`Wayland`**: Can only be set while building the window, setting the window's application ID.
+    /// - **`X11`**: Can only be set while building the window, setting the window's `WM_CLASS`.
+    /// - **`iOS`**, **`Android`**, and **`Web`**: not applicable.
+    ///
+    /// Notes: Changing this field during runtime will have no effect for now.
+    pub app_id: Option<String>,
```

Comment:
I would prefer `name` rather than `app_id` as it's the name used by winit

Reply:
Initially, I leaned towards using `app_id` for a couple of reasons:
1. In the Linux (or Wayland) community, which forms the primary audience for this PR, `app_id` is a more commonly recognized term. Opting for `name` as the field name might reduce its discoverability, as `name` is quite a generic term and may not convey specific meaning effectively.
2. In Wayland, the application ID serves a role similar to the `WM_CLASS` in X11. It's worth noting that X11 also uses another property, `WM_NAME`, but that serves a different purpose... Oh, really? 

However, upon further reflection, I've concluded that `name` is actually a more suitable choice. The title of a window is intended for users, whereas the window's `name`, irrespective of whether it's referred to as the application ID, `WM_CLASS`, `WM_NAME`, or window class name, is used by the window manager or compositor to categorize and manage windows according to specific windowing rules.